client/src/components/LeafletOsmMap/index.js

```diff
@@ -69,26 +73,80 @@ export default class LeafletOSMMap extends Component {
 
   /**
    * Handle leaflet map get device location event
-   * @param {*} event 
+   * @param {*} event
    */
   onLocationFound(event) {
     this.state.map.setZoom(16);
     const userMarker = L.circleMarker(event.latlng, {
       radius: 8,
       weight: 3,
-      fillColor: 'red',
-    }).addTo(this.state.map)
+      fillColor: 'red'
+    })
+      .addTo(this.state.map)
       .bindPopup('You Are Here');
 
     this.setState({
       mapCenter: event.latlng,
-      userMarker: userMarker,
+      userMarker: userMarker
+    });
+
+    L.Control.Center = L.Control.extend({
+      onAdd: function(map) {
+        let container = L.DomUtil.create(
+          'div',
+          'leaflet-bar leaflet-control leaflet-control-custom'
+        );
+
+        let controlText = L.DomUtil.create('div');
+        controlText.style.color = 'rgb(25,25,25)';
+        controlText.style.fontFamily = 'Roboto,Arial,sans-serif';
+        controlText.style.fontSize = '16px';
+        controlText.style.lineHeight = '38px';
+        controlText.style.paddingLeft = '5px';
+        controlText.style.paddingRight = '5px';
+        controlText.innerHTML = 'Center Map';
+        container.appendChild(controlText);
+
+        L.DomEvent.disableClickPropagation(container);
+
+        container.style.backgroundColor = '#fff';
+        container.style.border = '2px solid #e7e7e7';
+        container.style.borderRadius = '3px';
+        container.style.boxShadow = '0 2px 6px rgba(0,0,0,.3)';
+        container.style.cursor = 'pointer';
+        container.style.marginBottom = '22px';
+        container.style.textAlign = 'center';
+        container.title = 'Click to recenter the map';
+
+        L.DomEvent.on(container, 'mouseenter', function(e) {
+          e.target.style.background = '#e7e7e7';
+        });
+
+        L.DomEvent.on(container, 'mouseleave', function(e) {
+          // e.target.style.color = '#ccc';
+          e.target.style.background = '#fff';
+        });
+
+        L.DomEvent.on(container, 'click', function() {
+          map.flyTo(map._lastCenter);
```

Comment:
This might be the culprit of your bug. Rather, you want to fly back to use location. You can get the latlng date from the `state.userMarker`. This object is what we're using to track user location and will continuously be updated as the user moves.